apps/vc-api/src/vc-api/credentials/credentials.service.ts

```diff
@@ -177,12 +166,14 @@ export class CredentialsService implements CredentialVerifier {
     vp: VerifiablePresentationDto,
     options: VerifyOptionsDto
   ): Promise<VerificationResultDto> {
-    const verifyOptions: ISpruceVerifyOptions = options;
-
-    return didKitExecutor<VerificationResultDto>(
-      () => verifyPresentation(JSON.stringify(vp), JSON.stringify(verifyOptions)),
-      'verifyPresentation'
+    const w3cVerifyPresentationOptions: W3cVerifyPresentationOptions = {
+      presentation: JsonTransformer.fromJSON(vp, W3cJsonLdVerifiablePresentation),
```

Comment:
As discussed during our call on Friday, presentation verification is failing if the presentation doesn't have a `verifiableCredential` property (as is the case with the `authentication` presentations). I propose that what we do is skip Credo's `fromJSON` validation in this one case. From my testing, this allows `authentication` presentations to be verified.
```suggestion
    const w3cVerifyPresentationOptions: W3cVerifyPresentationOptions = {
      presentation: JsonTransformer.fromJSON(vp, W3cJsonLdVerifiablePresentation, {
        // Credo is expecting the the verifiableCredential property in a VP
        // However, this property is optional in both the v1.1 and v2 VC data models
        // For example, it isn't present if the VP is only for authentication
        // Credo issue to resolve underlying cause: https://github.com/openwallet-foundation/credo-ts/issues/2038
        validate: !(vp.verifiableCredential === undefined)
      }),
```

Reply:
A note that as this `vp.verifiableCredential === undefined` exception needs to be handles in several places (3 in total I believe), I think it would probably make sense to extract the `fromJSON` call for VP into a reusable function (perhaps a private method in the service class)